src/dg/weak_dg.cpp

```diff
@@ -15,63 +15,71 @@
 
 #include "ADTypes.hpp"
 
+#include "solution/local_solution.hpp"
 #include "weak_dg.hpp"
 
 #define KOPRIVA_METRICS_VOL
 #define KOPRIVA_METRICS_FACE
 #define KOPRIVA_METRICS_BOUNDARY
 //#define FADFAD
 
+namespace {
+template <typename real, int dim> using Coord = std::array<real, dim>;
+// First index corresponds to the component of the coordinate, second index corresponds to the component of the gradient.
+template <typename real, int dim> using CoordGrad = std::array<dealii::Tensor<1, dim, real>, dim>;
+
+template <typename real, int nstate> using State = std::array<real, nstate>;
+// First index corresponds to the component of the state, second index corresponds to the component of the gradient.
+template <typename real, int dim, int nstate> using DirectionalState = std::array<dealii::Tensor<1, dim, real>, nstate>;
+}
+
+namespace {
```

Comment:
It's called an anonymous namespace. Those free functions are currently only used inside `weak_dg.cpp`. By putting it into an anonymous namespace, we guarantee that it will not clash with any other functions with the same name in PHiLiP. The functions in here were originally declared/defined in `weak_dg.cpp` and are not used anywhere else.
https://stackoverflow.com/a/357464

I considered migrating them to a different file like `dg_helper_functions.cpp` to use it for `strong_dg.cpp`. However, `StrongDG` is just its own entity at this point. There needs to be some major refactoring so that they play nice together.

Reply:
Thanks for the explanation!